walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,63 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+
 
 public final class FindMeetingQuery {
+  /**Return all possible time intervals for a meeting, taking into account the optional attendees, 
+   if it's possible, otherwise return all the possible time intervals without them.*/
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    ArrayList<TimeRange> meetingsWithOptional = findOptionalTimeForMeeting(events, request, true);
```

Comment:
Prefer to use the Guava immutable types (com.google.common.collect.ImmutableList, etc.) when possible, instead of the native Java collections. See go/ej3e-17.

Reply:
If I return an immutable List, do I need to change all the tests accordingly? Is there another way?